lib/src/main/java/com/auth0/android/lock/utils/CustomField.java

```diff
@@ -56,22 +58,68 @@
         int TYPE_EMAIL = 3;
     }
 
+    /**
+     * Location to store the field into.
+     */
+    @IntDef({PROFILE_ROOT, USER_METADATA})
+    @Retention(RetentionPolicy.SOURCE)
+    public @interface Storage {
+        /**
+         * Store the field into the user profile root.
+         * The list of attributes that can be added to your root profile is here https://auth0.com/docs/api/authentication#signup.
+         */
+        int PROFILE_ROOT = 0;
+
+        /**
+         * Store the field into the user_metadata object.
+         */
+        int USER_METADATA = 1;
+    }
+
     @DrawableRes
-    private int icon;
+    private final int icon;
     @FieldType
     private final int type;
     private final String key;
     @StringRes
     private final int hint;
-
+    @Storage
+    private final int storage;
+
+
+    /**
+     * Constructor for CustomField instance
+     *
+     * @param icon the icon resource to display next to the field.
+     * @param type the type of input this field will receive. Used to determine the applied validation.
+     * @param key  the key to store this field as.
+     * @param hint the placeholder to display when this field is empty.
+     */
     public CustomField(@DrawableRes int icon, @FieldType int type, @NonNull String key, @StringRes int hint) {
+        this(icon, type, key, hint, USER_METADATA);
+    }
+
+    /**
+     * Constructor for CustomField instance
+     *
+     * @param icon    the icon resource to display next to the field.
+     * @param type    the type of input this field will receive. Used to determine the applied validation.
+     * @param key     the key to store this field as.
+     * @param hint    the placeholder to display when this field is empty.
+     * @param storage the location to store this value into. Defaults to {@link Storage#USER_METADATA}.
```

Comment:
How does this default to `Storage.USER_METADATA`? I see the other constructor will pass in `USER_METADATA`, but since this is also a public constructor, there really isn't a "default" here is there?

Reply:
I agree. I'll clarify that the default applies to the other constructor only.